test/core/currency/confirmations.test.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+
+import { expect } from 'chai'
+import { describe, it } from 'mocha'
+
+import { validateConfirmations } from '../../../src/core/currency/wallet/currency-wallet-callbacks'
+
+describe('confirmations API', function () {
+  const helper = (
+    txBlockHeight: number,
+    netBlockHeight: number,
+    required: number,
+    expected: string
+  ): void => {
+    const tx: any = { blockHeight: txBlockHeight }
+    expect(
+      validateConfirmations(tx, netBlockHeight, required),
+      `Expected tx with blockHeight of ${tx.blockHeight} to be ${expected} at network blockHeight ${netBlockHeight} with ${required} required confs`
+    ).equals(expected)
+  }
+  const loop = (start: number, end: number, fn: (v: number) => void) => {
+    for (let i = start; i <= end; ++i) {
+      fn(i)
+    }
```

Comment:
I'm not sure these abstractions help that much. Unit tests should be obvious & dumb. I don't think you need to fix / re-write these tests, since they work, but they are harder to read & understand like this.